integration_create_delete_test.go

```diff
@@ -189,18 +301,56 @@ func waitForSnapshotToBeDeleted(snapshotId string, svc *ec2.EC2, logger *log.Log
 	time.Sleep(30 * time.Second)
 }
 
-func deleteSnapshotForInstance(instance *ec2.Instance, ui cli.Ui, logger *log.Logger, t *testing.T) {
-	logger.Printf("Deleting snapshot for instance %s", *instance.InstanceId)
+func deleteSnapshotForInstance(instanceName string, olderThan string, requireAtLeast int, ui cli.Ui, logger *log.Logger, t *testing.T) {
+	logger.Printf("Deleting snapshot for instance %s", instanceName)
 
 	deleteCmd := DeleteCommand{
 		Ui: ui,
 		AwsRegion: AWS_REGION_FOR_TESTING,
-		InstanceId: *instance.InstanceId,
-		OlderThan: "0h",
-		RequireAtLeast: 0,
+		InstanceName: instanceName,
+		OlderThan: olderThan,
+		RequireAtLeast: requireAtLeast,
 	}
 
 	if err := deleteSnapshots(deleteCmd); err != nil {
 		t.Fatal(err)
 	}
+}
+
+func takeSnapshotWithVerification(instanceName string, instanceId string, ui cli.Ui, svc *ec2.EC2, logger *log.Logger, t *testing.T) string {
+	snapshotId := takeSnapshot(instanceName, ui, logger, t)
+
+	waitForSnapshotToBeAvailable(instanceId, svc, logger, t)
```

Comment:
Got a CircleCI failure in https://circleci.com/gh/josh-padnick/ec2-snapper/13?utm_campaign=build-failed&utm_medium=email&utm_source=notification of this:

```
TestDeleteRespectsAtLeast 2016/04/14 16:24:48 Terminating instance i-22a471bf
--- FAIL: TestDeleteRespectsAtLeast (734.28s)
	integration_create_delete_test.go:293: ResourceNotReady: exceeded 40 wait attempts
```

Probably just AWS variations and not worth changing anything. I'll re-run the build for you. https://circleci.com/gh/josh-padnick/ec2-snapper/15